if/src/lib/initialize.ts

```diff
@@ -0,0 +1,176 @@
+import pathLib = require('path');
+
+import {ERRORS} from '../util/errors';
+import {logger} from '../util/logger';
+
+import {CONFIG, STRINGS} from '../config';
+
+import {PluginInterface} from '../types/interface';
+import {PluginsStorage} from '../types/initialize';
+import {GlobalPlugins, PluginOptions} from '../types/manifest';
+import {resolve} from 'path';
+import {exec, ExecException} from 'child_process';
+import {getPluginsDirectoryPath} from '../util/helpers';
+
+const {ModuleInitializationError, PluginCredentialError} = ERRORS;
+
+const {GITHUB_PATH, NATIVE_PLUGINS} = CONFIG;
+const {
+  MISSING_METHOD,
+  MISSING_PATH,
+  NOT_NATIVE_PLUGIN,
+  INVALID_MODULE_PATH,
+  PLUGIN_INSTALL_CHECK_FAILED,
+} = STRINGS;
+
+interface CommandOutput {
+  dependencies?: Record<string, unknown>;
+}
+
+interface ExecError {
+  error: ExecException | null;
+  stderr: string;
+}
+
+const checkIfModuleIsInstalled = async (path: string) => {
+  try {
+    const stdout = await execAsync(
+      `npm list --include=prod --prefix=${getPluginsDirectoryPath()} --json`
+    );
+
+    let stdoutParsed: CommandOutput;
+    try {
+      stdoutParsed = JSON.parse(stdout);
+    } catch (error) {
+      logger.error('Error parsing stdout of command.');
+      return undefined;
+    }
+
+    return path in (stdoutParsed?.dependencies ?? {});
+  } catch (error) {
+    if (isExecError(error)) {
+      error.error &&
+        logger.error(
+          `Error during call to npm when checking if plugin is installed: ${error.error.message}`
+        );
+      error.stderr !== '' &&
+        logger.error(`Stderr from plugin install check: ${error.stderr}`);
+    } else {
+      logger.error(`Error while checking if plugin is installed: ${error}`);
+    }
+  }
+
+  return undefined;
+};
+
+/**
+ * Imports module by given `path`.
+ */
+const importModuleFrom = async (path: string) => {
+  const isModuleInstalled = await checkIfModuleIsInstalled(path);
```

Comment:
This doesn't seem like a good idea anymore. It just adds complexity and the current implementation opens up the tool to shell injection attacks (even though from a manifest file that you decide to execute yourself - but still unnecessary I think).

It should be enough to check if `resolve(pluginsDir, path)` is still in `plugins`.